tubemq-client-twins/tubemq-client-go/selector/ip_selector.go

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package selector
+
+import (
+	"errors"
+	"strings"
+)
+
+func init() {
+	s := &ipSelector{}
+	s.indexes = make(map[string]int)
+	Register("ip", s)
+	Register("dns", s)
+}
+
+type ipSelector struct {
```

Comment:
Can the following structure be used?
type ipSelector struct {	
indexes map[string]ipServices
}

type  ipServices struct {
nextIndex int
addresss []string
}

Reply:
Good idea.